### ISSUE TITLE
Feature Request: 'Dump WorkflowTask execution data as JSON'

### ISSUE THREAD

**Issue by ryantanaka** [wrench-project/wrench#72]
After #71 is complete:

- [x] Create a function to dump workflow task execution data as JSON 

- [x] Create a test

- [x] Change num_cores_used to num_cores_allocated

**Comment by ryantanaka** [comment]
This is complete and the changes have been merged into master.
